packages/fxa-auth-server/fxa-oauth-server/lib/db/mysql/index.js

```diff
@@ -438,7 +441,16 @@ MysqlStore.prototype = {
   getClients: function getClients(email) {
     return this._read(QUERY_CLIENT_LIST, [email]);
   },
-  removeClient: function removeClient(id) {
+  // TODO: beware, this is a very slow series of blocking deletes. We should
+  //       upgrade to a redis library that supports the UNLINK command, which
+  //       would instead do the deletes of all client tokens in a non-blocking
+  //       background thread. #2623
+  removeClient: async function removeClient(id) {
+    const clientId = id.toString('hex');
+    const clientTokenIds = await this.redis.smembers(clientId);
+    for (const tokenId of clientTokenIds) {
+      await this.removeAccessToken(tokenId);
```

Comment:
i think an unlink would be just as slow as a del on a single key. it's only when you pass multiple keys in the same call that unlink is faster

Reply:
:+1: Yeah, my concern with unlink is for deleting clients, where we'd need to delete every token stored under the client ID